nncf/experimental/torch/sparsity/movement/layers.py

```diff
@@ -0,0 +1,316 @@
+                raise ValueError(
+                    f'{error_prefix} Missing `axis`. Per-dim sparsity structure expects it specified.')
+            if sparse_factors is not None:
+                raise ValueError(
+                    f'{error_prefix} Per-dim sparsity structure does not expect specified `sparse_factors`.')
+            self.sparse_axis = int(sparse_axis)
+
+    @classmethod
+    def from_config(cls, config: Dict[str, Any]) -> 'SparseConfig':
+        """
+        Creates the object from its config.
+        """
+        mode_str = config.get('mode', SparseStructure.FINE.value)
+        mode = SparseStructure(mode_str)
+        sparse_factors = config.get('sparse_factors')
+        axis = config.get('axis')
+        return cls(mode, sparse_factors, axis)
+
+    def __str__(self) -> str:
+        return f'{self.mode.value, self.sparse_factors}'
+
+
+class SparseConfigByScope:
+    """
+    Defines an entry for `sparse_structure_by_scopes` in movement sparsity configuration.
+    It includes the sparse config, and the target scopes it is applied to.
+    """
+
+    def __init__(self, sparse_config: SparseConfig, target_scopes: Union[str, List[str]]):
+        self.sparse_config = sparse_config
+        self.target_scopes = target_scopes
+
+    @classmethod
+    def from_config(cls, config: Dict[str, Any]) -> 'SparseConfigByScope':
+        """
+        Creates the object from its representation.
+        """
+        error_prefix = f'Invalid sparse structure by scopes {config}.'
+        target_scopes = config.get('target_scopes')
+        if not target_scopes:
+            raise ValueError(f'{error_prefix} Missing `target_scopes`.')
+        sparse_config = SparseConfig.from_config(config)
+        return cls(sparse_config, target_scopes)
+
+
+@COMPRESSION_MODULES.register()
+class MovementSparsifier(nn.Module):
+    """
+    Defines the operand of movement sparsity for supported layers.
+    """
+
+    def __init__(
+        self,
+        target_module_node: NNCFNode,
+        sparse_cfg: SparseConfig = SparseConfig(mode=SparseStructure.FINE),
+        frozen: bool = True,
+        compression_lr_multiplier: Optional[float] = None,
+        layerwise_loss_lambda: float = 0.5,
+    ):
+        """
+        Initializes the operand of movement sparsity for a certain layer.
+
+        :param target_module_node: Node name of the module this operand is related to.
+        :param sparse_cfg: Sparse structure config for the module to sparsify.
+        :param frozen: Whether the operand is frozen, i.e., binary masks are fixed.
+        :param compression_lr_multiplier: The value of gradient multiplier for learnable parameters in the operand.
+        :param layerwise_loss_lambda: The extra factor of compression loss for this specific layer.
+        """
+        super().__init__()
+        self.target_module_node = target_module_node
+        self.prune_bias = (target_module_node.layer_attributes.bias not in (False, None))
```

Comment:
Is boolean type correct for bias? In other places bias checks only as `is None`.

If its ok, it can be changed to  `bool(target_module_node.layer_attributes.bias)`